packages/kbn-es-query/src/kuery/utils/escape_kuery.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { flow } from 'lodash';
+
+/**
+ * Escapes backslashes and double-quotes. (Useful when putting a string in quotes to use as a value
+ * in a KQL expression. See the QuotedCharacter rule in kuery.peg.)
+ */
+export function escapeQuotes(str: string) {
+  return str.replace(/[\\"]/g, '\\$&');
+}
+
+/**
+ * Escapes a Kuery node value to ensure that special characters, operators, and whitespace do not result in a parsing error or unintended
+ * behavior when using the value as an argument for the `buildNode` function.
+ */
+export const escapeKuery = flow(escapeSpecialCharacters, escapeAndOr, escapeNot, escapeWhitespace);
+
+// See the SpecialCharacter rule in kuery.peg
+function escapeSpecialCharacters(str: string) {
+  return str.replace(/[\\():<>"*]/g, '\\$&'); // $& means the whole matched string
```

Comment:
@elastic/kibana-data-discovery 

I have one question  regarding above Regex. This regex skips `{` & `}` which are included in the specs from `grammar.peggy` as shown below. Do we know why?

https://github.com/elastic/kibana/blob/d6b4fe9e6ee2e18691b0fa818c4e2c4ed3c413fc/packages/kbn-es-query/grammar/grammar.peggy#L295-L298


https://github.com/elastic/kibana/blob/d6b4fe9e6ee2e18691b0fa818c4e2c4ed3c413fc/packages/kbn-es-query/grammar/grammar.peggy#L295-L298

Reply:
Good catch, yeah they should be added to `escapeSpecialCharacters` since they're special syntax for querying nested fields. That util must not have been updated when that syntax was added.

Reply:
Okay Thanks. For now, I will leave this PR as it is and will raise a separate issue and PR for adding the missing characters.